### ISSUE TITLE
ESLint API rejects import/no-cycle maxDepth: Infinity during config serialization

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1213]
The ESLint-compatible API rejects the standard `import/no-cycle` option `maxDepth: Infinity` while accepting a finite depth. `Infinity` is the rule's normal unlimited-depth value in JavaScript configuration.

Reproduced with `@utoo/lint@0.4.0`, Node.js 22.23.1, macOS arm64.

### Reproduction

```sh
npm init -y
npm install @utoo/lint@0.4.0
```

Save as `repro.cjs` and run `node repro.cjs`:

```js
const { ESLint } = require('@utoo/lint');

(async () => {
  for (const maxDepth of [2, Infinity]) {
    const engine = new ESLint({
      useEslintrc: false,
      overrideConfig: {
        rules: {
          'import/no-cycle': ['error', { maxDepth }]
        }
      }
    });
    try {
      const [result] = await engine.lintText(
        'export const value = 1;',
        { filePath: 'input.js' }
      );
      console.log(String(maxDepth), result.errorCount);
    } catch (error) {
      console.log(String(maxDepth), error.message);
    }
  }
})();
```

### Actual

- `maxDepth: 2`: succeeds with zero errors.
- `maxDepth: Infinity`: throws `invalid flat config .../utlint.config.json: UnsupportedRuleConfigValue`.

### Expected

`Infinity` should preserve unlimited-depth semantics, or be normalized to the native representation of the same default before writing temporary JSON. This is a runtime JavaScript config value, not an attempt to put an Infinity literal in a JSON file.

The symptom is consistent with serializing `Infinity` to JSON `null`. An actionable diagnostic that names the rule and option would also help if a value cannot be represented safely.
